ci/release/update-version.sh

```diff
@@ -38,11 +103,22 @@ echo "${NEXT_FULL_TAG}" > VERSION
 
 sed_runner "s#cucim.kit.cuslide@${CURRENT_LONG_TAG}.so#cucim.kit.cuslide@${NEXT_FULL_TAG}.so#g" cucim.code-workspace
 
+# CI files - context-aware branch references
 for FILE in .github/workflows/*.yaml; do
-  sed_runner "/shared-workflows/ s/@.*/@branch-${NEXT_SHORT_TAG}/g" "${FILE}"
+  sed_runner "/shared-workflows/ s|@.*|@${RAPIDS_BRANCH_NAME}|g" "${FILE}"
   sed_runner "s/:[0-9]*\\.[0-9]*-/:${NEXT_SHORT_TAG}-/g" "${FILE}"
 done
 
+# Notebook references - context-aware
+if [[ "${RUN_CONTEXT}" == "main" ]]; then
+  # In main context, keep notebooks on main (no changes needed)
+  :
+elif [[ "${RUN_CONTEXT}" == "release" ]]; then
+  # In release context, use release branch for notebook links (word boundaries to avoid partial matches)
+  sed_runner "s|\\bmain\\b|release/${NEXT_SHORT_TAG}|g" notebooks/Using_Cache.ipynb
+  sed_runner "s|\\bmain\\b|release/${NEXT_SHORT_TAG}|g" notebooks/Supporting_Aperio_SVS_Format.ipynb
+fi
```

Comment:
Recently caught a bug: https://github.com/rapidsai/cucim/pull/972#discussion_r2586404752

This has me questioning these lines

Could you please share a bit more about why these were added?